hack/devtools/rp_dev_helper.sh

```diff
@@ -1,39 +1,471 @@
+        abort "'skip_deployments' was set to 'false'. ❌⏩ Don't skip certs import to keyVault $keyVault"
+    fi
+
+    for i in "${!certificates[@]}"; do
+         if check_keyvault_certificate "${keyVault}" "${certificates[i]}" $skip_deployments; then
+            log "⏭️🔑💼 Skip import for certificate ${certificates[i]}"
+        else
+            log "🔐📥 Import certificate ${certificates[i]}"
+            az keyvault certificate import \
+                --vault-name "${keyVault}" \
+                --name "${certificates[i]}" \
+                --file "${secret_files[i]}" >/dev/null
+        fi
+    done
+}
+
+check_and_import_certificates (){
+    err_str="Usage $0 <KEYVAULT_PREFIX> [SKIP_DEPLOYMENTS]. Please try again"
+    local keyvault_prefix=${1?$err_str}
+    local skip_deployments=${2:-"true"}
+
+    # Don't skip deployment creation when skip_deployments was set to "false" 
+    if is_it_boolean $skip_deployments && ! $skip_deployments; then
+        abort "'skip_deployments' was set to 'false'. ❌⏩ Don't skip certs import"
+    fi
+
+    local files
+    local certs
+    local key_vault
+    key_vault="${keyvault_prefix}-svc"
+    log "Check import certificates for the service keyVault ${key_vault}"
+    certs=("rp-mdm" "rp-mdsd" "cluster-mdsd" "dev-arm" "rp-firstparty" "rp-server")
+    files=("secrets/rp-mdm-self-signed.pem" "secrets/rp-mdsd-self-signed.pem" "secrets/cluster-mdsd-self-signed.pem" "secrets/arm.pem" "secrets/firstparty.pem" "secrets/localhost.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+    
+    key_vault="${keyvault_prefix}-gwy"
+    log "Check import certificates for the gateway keyVault ${key_vault}"
+    certs=("gwy-mdm" "gwy-mdsd")
+    files=("secrets/gwy-mdm-self-signed.pem" "secrets/gwy-mdsd-self-signed.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+
+    key_vault="${keyvault_prefix}-por"
+    log "Check import certificates for the portal keyVault ${key_vault}"
+    certs=("portal-server" "portal-client")
+    files=("secrets/localhost.pem" "secrets/portal-client.pem")
+    skip_and_import_certificates ${key_vault} ${certs[@]} ${files[@]} $skip_deployments
+}
+
+clean_rp_dev_env() {
+    err_str="Usage $0 <LOCATION> [LIST_RESOURCE_GROUPS] [LIST_KEYVAULTS]. Please try again"
+    local location=${1?$err_str}
+    local -a list_resource_groups=("${2:-}")
+    local -a list_keyvaults=("${3:-}")
+    log "########## Deleting Dev Env in $location ##########"
+
+    if [[ $# -lt 2 ]]; then
+        log "Info: One input argument was required. Checking AZURE_PREFIX env var for default values"
+        if [[ -z "${AZURE_PREFIX}" ]]; then
+            abort "Error: AZURE_PREFIX environment variable was not set."
+        fi
+        log "Info: AZURE_PREFIX=${AZURE_PREFIX}"
+    fi
+
+    # Convert input strings to arrays
+    eval "rgs=($list_resource_groups)"
+    eval "kvs=($list_keyvaults)"
```

Comment:
1. Does this work? You are expanding arrays without an index, so you will only get the first element.
2. Is this nessecary? you are already declaring and assigning them as arrays on lines 338 and 339.

Reply:
>Does this work? You are expanding arrays without an index, so you will only get the first element.

It does.

>Is this nessecary? you are already declaring and assigning them as arrays on lines 338 and 339.

It is necessary when you want to explicitly delete an array or resource groups and an array of keyVaults. Otherwise, the function will try these kinds of resources based on default values.